### PR TITLE
enhance(dep): adds dep check in travis

### PR DESCRIPTION
**Pull request by ashishranjan738**
Signed-off-by: Ashish Ranjan <ashishranjan738@gmail.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This commit adds `dep check` in travis which will allow catching broken dependencies PR.

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes-sigs/azuredisk-csi-driver/issues/54

**Special notes for your reviewer**:
Merge this PR after https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/64 as curretly the vendor packages are not in sync and the PR #64 contains the changes of `dep ensure`.

**Release note**:
```
adds dep check in travis
```

### CODE REVIEW

File: hack/verify-dep.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+echo "Verifying dep check"
+dep check
```
Comment (andyzhangx) [inline hack/verify-dep.sh]: if `dep check` failed, should it `exit $errorno`? otherwise travis ci still passes?

Comment (ashishranjan738) [reply hack/verify-dep.sh]: `dep check` returns non zero code if dependencies are not satisfied, which will make Travis exit. We don't to check for return code explicitly.

**Comment by andyzhangx** [comment]
would you also fix this failure in this PR:
```
Verifying dep check
# Gopkg.lock is out of sync:
github.com/Azure/go-autorest/autorest/to: imported or required, but missing from Gopkg.lock's input-imports
# vendor is out of sync:
google.golang.org/api: hash of vendored tree not equal to digest in Gopkg.lock
k8s.io/kubernetes: hash of vendored tree not equal to digest in Gopkg.lock
The command "hack/verify-all.sh" exited with 1.
```

**Comment by ashishranjan738** [comment]
> would you also fix this failure in this PR:
> 
> ```
> Verifying dep check
> # Gopkg.lock is out of sync:
> github.com/Azure/go-autorest/autorest/to: imported or required, but missing from Gopkg.lock's input-imports
> # vendor is out of sync:
> google.golang.org/api: hash of vendored tree not equal to digest in Gopkg.lock
> k8s.io/kubernetes: hash of vendored tree not equal to digest in Gopkg.lock
> The command "hack/verify-all.sh" exited with 1.
> ```

@andyzhangx fixed

**Comment by andyzhangx** [comment]
thanks, could you also fix it in azure file driver: https://github.com/kubernetes-sigs/azurefile-csi-driver/issues/39, thanks.

**Pull request merged**